com.minres.coredsl.tests/src/com/minres/coredsl/tests/CoreDslElaborationTest.xtend

```diff
@@ -159,19 +94,25 @@ class CoreDslElaborationTest {
 					int x = 32;
 				}
 			}
-		''', error(IssueCodes.DuplicateIsaStateElement), warning(IssueCodes.DuplicateIsaStateElement));
+		'''.testProgram()
+		.expectError(IssueCodes.DuplicateIsaStateElement)
+		.expectWarning(IssueCodes.DuplicateIsaStateElement)
```

Comment:
Why is one of the decls (only) a warning?

Reply:
The warning is for the more general case and will be emitted whenever two ISA parameters with the same name are visible for one ISA. This isn't necessarily an error, because it is valid if the declarations are in disjoint instruction sets provided by a core.

The error is for the more specific violation where one of the two declarators can see the other  i.e. they are declared in the same hierarchy.

I would expect the error to be reported on the second declarator and the warning on the ISA name, but I'm not 100% sure on that.

Reply:
Just checked and that is exactly what happens:
![image](https://user-images.githubusercontent.com/4553050/220397811-95925429-080e-4165-92a7-c730efd7569a.png)